episodes/vaccine-comparisons.Rmd

```diff
@@ -0,0 +1,968 @@
+output_vaccinate_01$intervention_type  <- "vaccinate 0.001"
+output_vaccinate_02$intervention_type  <- "vaccinate 0.01"
+output <- rbind(output_baseline, output_vaccinate_01, output_vaccinate_02)
+
+
+output %>%
+  filter(compartment == "vaccinated") %>%
+  ggplot() +
+  ggtitle("Number vaccinated") +
+  aes(
+    x = time,
+    y = value,
+    color = intervention_type,
+    linetype = intervention_type
+  ) +
+  stat_summary(
+    fun = "sum",
+    geom = "line",
+    linewidth = 1
+  ) +
+  scale_y_continuous(
+    labels = scales::comma
+  ) +
+  geom_vline(
+    xintercept = c(
+      vaccinate_01$time_begin,
+      vaccinate_01$time_end
+    ),
+    linetype = 2
+  ) +
+  theme_bw() +
+  labs(
+    x = "Simulation time (days)",
+    y = "Individuals"
+  )
+```
+
+### R Code
+
+```{r, echo = TRUE, eval = FALSE}
+output_baseline <- model_default(
+  # population
+  population = uk_population,
+  # rate
+  transmission_rate = transmission_rate,
+  infectiousness_rate = infectiousness_rate,
+  recovery_rate = recovery_rate,
+  # time
+  time_end = 200, increment = 1.0
+)
+
+output_vaccinate_01 <- model_default(
+  # population
+  population = uk_population,
+  # rate
+  transmission_rate = transmission_rate,
+  infectiousness_rate = infectiousness_rate,
+  recovery_rate = recovery_rate,
+  # intervention
+  vaccination = vaccinate_01,
+  # time
+  time_end = 200, increment = 1.0
+)
+
+output_vaccinate_02 <- model_default(
+  # population
+  population = uk_population,
+  # rate
+  transmission_rate = transmission_rate,
+  infectiousness_rate = infectiousness_rate,
+  recovery_rate = recovery_rate,
+  # intervention
+  vaccination = vaccinate_02,
+  # time
+  time_end = 200, increment = 1.0
+)
+
+# create intervention_type column for plotting
+output_baseline$intervention_type <- "baseline"
+output_vaccinate_01$intervention_type <- "vaccinate 0.001"
+output_vaccinate_02$intervention_type <- "vaccinate 0.01"
+output <- rbind(output_baseline, output_vaccinate_01, output_vaccinate_02)
+
+
+output %>%
+  filter(compartment == "vaccinated") %>%
+  ggplot() +
+  ggtitle("Number vaccinated") +
+  aes(
+    x = time,
+    y = value,
+    color = intervention_type,
+    linetype = intervention_type
+  ) +
+  stat_summary(
+    fun = "sum",
+    geom = "line",
+    linewidth = 1
+  ) +
+  scale_y_continuous(
+    labels = scales::comma
+  ) +
+  geom_vline(
+    xintercept = c(
+      vaccinate_01$time_begin,
+      vaccinate_01$time_end
+    ),
+    linetype = 2
+  ) +
+  theme_bw() +
+  labs(
+    x = "Simulation time (days)",
+    y = "Individuals"
+  )
+```
+
+
+:::
+
+To show the **indirect** effect of vaccinations, we want to see the effect that vaccination has on the number of new infections. We will use the function `new_infections()` in `{epidemics}` to calculate the number of new infections over time for the different vaccination programs. 
+
+The inputs required are :
+
++ `data` : the model output,
```

Comment:
Beyond scope of this specific PR, but wonder if it's unclear to learners to have epidemics models generate an output labelled 'data' for simulations?